web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcDirective.js

```diff
@@ -526,28 +527,58 @@
 
 
                 function loadLayers() {
-                  if (!angular.isArray(scope.map.getSize()) ||
-                      scope.map.getSize().indexOf(0) >= 0) {
-                    $timeout(function() {
-                      scope.map.updateSize();
-                      if (projectedExtent != null) {
-                        scope.map.getView().fit(
-                          projectedExtent,
-                          scope.map.getSize());
+                  scope.map.get('creationPromise').then(function() {
+                    if (!angular.isArray(scope.map.getSize()) ||
+                        scope.map.getSize().indexOf(0) >= 0) {
+                      $timeout(function() {
+                        scope.map.updateSize();
+                        if (projectedExtent != null) {
+                          scope.map.getView().fit(
+                            projectedExtent,
+                            scope.map.getSize());
+                        }
+                      }, 300);
+                    }
+                  })
+
+                  /**
+                   * Function used to remove a parameter (case insensitive) from a URL by parameter name
+                   * @param key the parameter name
+                   * @param sourceURL the original URL
+                   * @returns {string} a URL without the parameter called `key`.
+                   *
+                   * @see https://stackoverflow.com/a/16941754/1140558
+                   */
+                  function removeParam(key, sourceURL) {
```

Comment:
Can be used this https://github.com/geonetwork/core-geonetwork/blob/5b3457bc69c8e5bdfca95c606aece11558e3c8f4/web-ui/src/main/resources/catalog/components/UrlUtilsService.js#L54 ?

Reply:
If that method cb an remove a parameter from a URL case-independent then yes.

Reply:
I've removed the custom method and used `gnURLUtils.remove` instead.